src/integration-test/java/com/commercetools/sync/integration/services/impl/StateServiceImplIT.java

```diff
@@ -161,6 +161,7 @@ void fetchMatchingStatesByKeys_WithAnyExistingKeys_ShouldReturnASetOfStates() {
     assertThat(transition.getId()).isEqualTo(transitionState.getId());
     assertThat(transition.getObj()).isNull();
 
+    clearTransitions(CTP_TARGET_CLIENT, fetchState);
```

Comment:
Please check if this line is necessary, came with the Merge.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/702/commits/40a1359c55894abc96c98ec084a7b2f9b4268c9b
Thanks for noticing this. I had to do workaround to fix the tests. But after Andi's changes, this is resolved. So, reverted back to the old code.